source/PlayerInfo.cpp

```diff
@@ -2660,6 +2765,17 @@ void PlayerInfo::RegisterDerivedConditions()
 	flagshipPlanetProvider.SetHasFunction(flagshipPlanetFun);
 	flagshipPlanetProvider.SetGetFunction(flagshipPlanetFun);
 
+	// Read only exploration conditions.
+	auto &&visitedPlanetProvider = conditions.GetProviderPrefixed("visited planet: ");
+	auto visitedPlanetFun = [this](const string &name) -> bool { return HasVisited(*GameData::Planets().Get(name)); };
```

Comment:
Also here and for systems, if the Planet doesn't exist, then this function should not create it.